struqture/src/mixed_systems/mixed_open_system.rs

```diff
@@ -114,6 +114,177 @@ impl<'a> OpenSystem<'a> for MixedLindbladOpenSystem {
                 actual_number_fermion_subsystems: noise.number_fermions.len(),
             });
         }
+
+        let mut current_number_spins = system.number_spins.clone();
+        let noise_number_spins = noise.number_spins.clone();
+        let noise_number_current_spins = noise.number_spins();
+        let system_number_current_spins = system.number_spins();
+        for (index, (system_spins, noise_spins)) in current_number_spins
+            .iter_mut()
+            .zip(noise_number_spins.iter())
+            .enumerate()
+        {
+            if system_spins != noise_spins {
+                match (*system_spins, noise_spins) {
+                    (Some(n), None) => {
+                        if n < noise_number_current_spins[index] {
+                            return Err(StruqtureError::MissmatchedNumberSubsystems {
```

Comment:
It looks like here the error is raised if the number of spins in a particular subsystem are missmatched, but the error just shows the number of subsystems of each type, which might as well be the same between the system and noise parts

Reply:
Fixed